### PR TITLE
Pathplanning

### PR DESCRIPTION
**Pull request by noahheller**

### CODE REVIEW

File: src/main/java/frc/robot/Robot.java [hunk 1]
```diff
@@ -5,16 +5,22 @@
...
 import frc.robot.commands.ResetGyro;
+import frc.robot.commands.SetInitOdom;
 import frc.robot.commands.drive.WheelAlign;
 
 public class Robot extends TimedRobot {
-    private RobotContainer robotContainer;
+    private final RobotContainer robotContainer;
+    private Command autoCommand;
+    public Robot (){
+        robotContainer = new RobotContainer();
```
Comment (armadilloz) [inline src/main/java/frc/robot/Robot.java]: Is this the way this needs to be done or one of the interim experiments? I think this is different than the way WPIlib is configured natively.

File: src/main/java/frc/robot/Robot.java [hunk 2]
```diff
@@ -29,4 +35,17 @@ public void testInit() {
         // Cancels all running commands at the start of test mode.
         CommandScheduler.getInstance().cancelAll();
     }
+
+    @Override
+    public void autonomousInit() {
+        autoCommand = robotContainer.getAutoCommand();
+        if (autoCommand!=null){
+            autoCommand.schedule();
```
Comment (armadilloz) [inline src/main/java/frc/robot/Robot.java:41]: I think there is also a cancel of the scheduled commands when switching to tele. can you check last year's code?

File: src/main/java/frc/robot/RobotContainer.java [hunk 3]
```diff
@@ -5,69 +5,99 @@
...
+    private Joystick joyleft = new Joystick(Constants.LEFT_JOYSICK_ID);
+    private Joystick joyright = new Joystick(Constants.RIGHT_JOYSTICK_ID);
+    private SwerveDrivetrain drivetrain;
+
+    private final ExampleAutoChooser autoChooser;
+
+    public RobotContainer() {
+        setupDriveTrain();
+        setupPathPlaning();
+        autoChooser = new ExampleAutoChooser();
```
Comment (armadilloz) [inline src/main/java/frc/robot/RobotContainer.java]: Are we using the `ExampleAutoChooser`?

File: src/main/java/frc/robot/RobotContainer.java [hunk 4]
```diff
@@ -5,69 +5,99 @@
...
+        autoChooser.addOnValidationCommand(()->new SetInitOdom(drivetrain,autoChooser));
```
Comment (armadilloz) [inline src/main/java/frc/robot/RobotContainer.java:50]: What is this doing?

File: src/main/java/frc/robot/RobotContainer.java [hunk 5]
```diff
@@ -5,69 +5,99 @@
...
-    drivetrain.resetOdometry(new Pose2d(0,0,new Rotation2d(Math.toRadians(0))));
-    
-    // Configure the trigger bindings
-    ramp = new Ramp();
-    configureBindings();
-    putShuffleboardCommands();
-
-  }
-  public void putShuffleboardCommands() {
-    SmartShuffleboard.putCommand("Ramp", "SetArmPID400", new RampMove(ramp, 400));
```
Comment (armadilloz) [inline src/main/java/frc/robot/RobotContainer.java:57]: These are missing from the new code?

File: src/main/java/frc/robot/autochooser/AutoAction.java [hunk 6]
```diff
@@ -1,60 +1,26 @@
...
-     }
-
-     @Override
-     public String toString() {
-          return getName();
-     }
+    DoNothing("Do Nothing"),
+    DIAGROT("DIAG ROT"),
+
+    FigureEight("Figure 8"),
```
Comment (armadilloz) [inline src/main/java/frc/robot/autochooser/AutoAction.java]: Remove?

File: src/main/java/frc/robot/autochooser/chooser/ExampleAutoChooser.java [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+import com.pathplanner.lib.path.PathPlannerPath;
+import edu.wpi.first.wpilibj2.command.Command;
+import frc.robot.autochooser.AutoAction;
+import frc.robot.autochooser.FieldLocation;
+import frc.robot.autochooser.PlaceHolderCommand;
+import frc.robot.autochooser.event.AutoEvent;
+
+import java.util.Map;
+
+public class ExampleAutoChooser extends Nt4AutoValidationChooser {
```
Comment (armadilloz) [inline src/main/java/frc/robot/autochooser/chooser/ExampleAutoChooser.java]: rename this class ?

File: src/main/java/frc/robot/commands/drive/WheelAlign.java [hunk 8]
```diff
@@ -4,21 +4,31 @@
 import edu.wpi.first.wpilibj.Timer;
 import edu.wpi.first.wpilibj2.command.Command;
 import frc.robot.Constants;
+import frc.robot.RobotContainer;
 import frc.robot.subsystems.swervev2.SwerveDrivetrain;
 
 public class WheelAlign extends Command {
     private final SwerveDrivetrain drivetrain;
+    private final double degree;
```
Comment (armadilloz) [inline src/main/java/frc/robot/commands/drive/WheelAlign.java]: What is the degree for? Does not seem to get used?

**Review by armadilloz — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/Robot.java (hunk 1 above)
Comment (noahheller) [reply src/main/java/frc/robot/Robot.java]: done

File: src/main/java/frc/robot/Robot.java (hunk 2 above)
Comment (noahheller) [reply src/main/java/frc/robot/Robot.java:41]: done

File: src/main/java/frc/robot/RobotContainer.java (hunk 3 above)
Comment (noahheller) [reply src/main/java/frc/robot/RobotContainer.java]: renamed

File: src/main/java/frc/robot/autochooser/AutoAction.java (hunk 6 above)
Comment (noahheller) [reply src/main/java/frc/robot/autochooser/AutoAction.java]: done

File: src/main/java/frc/robot/autochooser/chooser/ExampleAutoChooser.java (hunk 7 above)
Comment (noahheller) [reply src/main/java/frc/robot/autochooser/chooser/ExampleAutoChooser.java]: done

File: src/main/java/frc/robot/commands/drive/WheelAlign.java (hunk 8 above)
Comment (noahheller) [reply src/main/java/frc/robot/commands/drive/WheelAlign.java]: removed (left over from testing)

File: src/main/java/frc/robot/RobotContainer.java (hunk 5 above)
Comment (noahheller) [reply src/main/java/frc/robot/RobotContainer.java:57]: yes

**Review by armadilloz — APPROVED** [review]

**Pull request merged**